### PR TITLE
New extension: Perplexity-API

### PR DESCRIPTION
**Pull request by maximilianzuern**
## Description

This extension provides user access to all models through the [Perplexity API](https://docs.perplexity.ai/docs/getting-started).

There is a Raycast [perplexity extension](https://github.com/raycast/extensions/tree/main/extensions/perplexity) but it just opens query links in the browser for faster access to Perplexity web interface. 

This extension, though, gives user access to open-source models not offered on the web interface. That's why I built a new extension rather than updating the existing one because they serve different purposes.

### Selected Text Commands

- Change tone to casual
- Change tone to friendly
- Change tone to professional
- Comment code
- Compose Message/Mail responses
- Explain selected text
- Fix grammar and spelling
- Improve selected text
- Make selected text longer
- Make selected text shorter
- Summarize selected text
- Find synonyms according to the selected word
- Break down project description into tasks
- Translate selected text

### Quick AI
- Ask AI a one-shot question

### Supported Models

* codellama-34b-instruct
* codellama-70b-instruct
* llama-2-70b-chat
* mistral-7b-instruct
* mixtral-8x7b-instruct
* pplx-7b-chat
* pplx-70b-chat

## Screencast

prompt
askAI
fix
summary

Additionally, it calculates the cost per request, as displayed in the toast notification.

## Checklist

- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by pernielsentikaer** [comment]
I really think it would make sense to only have one Perplexity extension - what do you think, @third774?

This seems like a far better way to work with perplexity

**Comment by third774** [comment]
I'm hopeful that one day my extension can be deprecated/removed — especially with the upcoming work y'all are doing to directly integrate Perplexity with Raycast. That said, I would argue that the current extension is still _very_ useful in a few ways:
- No API key required!
- Can make use of Perplexity's copilot mode
- Can make use of Perplexity's focus modes

I don't know if these are available via the API access to models are they? Often, I also like to inspect the search results that are surfaced in the web UI. I think we'd be losing that as well, wouldn't we?

I'm def not opposed to merging these two extensions though if y'all think that's best! It's just that personally I think I'm going to keep using the web UI until those features are available directly in Raycast.

**Comment by pernielsentikaer** [comment]
What about we merge them so we can keep both yours and all the new stuff? It will require that content of the PR is added on top of the current extension so it's "just" an update 🙂

**Comment by maximilianzuern** [comment]
>     * No API key required!
> 
>     * Can make use of Perplexity's copilot mode
> 
>     * Can make use of Perplexity's focus modes
> 
> 
> I don't know if these are available via the API access to models are they? Often, I also like to inspect the search results that are surfaced in the web UI. I think we'd be losing that as well, wouldn't we?

Exactly! None of those features are accessible through the API. Moreover, the web UI uses models from OpenAI, Anthropic and Google, whereas the API relies on models from Meta, Mistral, and Perplexity.

> What about we merge them so we can keep both yours and all the new stuff? It will require that content of the PR is added on top of the current extension so it's "just" an update 🙂

The point is, the two extensions serve different purposes. Individuals seeking access to GPT3.5 or GPT4 prefer the web UI, whereas those who favor open-source models would use my extension.
Still, I get why you might consider merging them. In this case, should I modify the PR?

**Comment by pernielsentikaer** [comment]
Sounds like it's better to keep it as two separate, as long as the naming clearly shows it 🙂

File: extensions/perplexity-api/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# perplexity-api Changelog
```
Comment (pernielsentikaer) [inline extensions/perplexity-api/CHANGELOG.md:1]:
```suggestion
# Perplexity API Changelog
```

File: extensions/perplexity-api/README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# perplexity-api
```
Comment (pernielsentikaer) [inline extensions/perplexity-api/README.md:1]:
```suggestion
# Perplexity API
```

File: extensions/perplexity-api/package.json [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+          "type": "text",
+          "required": false
+        }
+      ]
+    }
+  ],
+  "preferences": [
+    {
+      "description": "Perplexity API Key",
+      "name": "apiKey",
```
Comment (pernielsentikaer) [inline extensions/perplexity-api/package.json:166]: Maybe add a `placeholder`

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you 

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/perplexity-api/package.json (hunk 3 above)
Comment (maximilianzuern) [reply extensions/perplexity-api/package.json:166]: Added via 587fe7f

**Comment by pernielsentikaer** [comment]
Would it make sense to make some of the commands disabled by default? Like we do in our [Spotify](https://www.raycast.com/mattisssa/spotify-player) extension 🙂

**Comment by maximilianzuern** [comment]
> Would it make sense to make some of the commands disabled by default? Like we do in our [Spotify](https://www.raycast.com/mattisssa/spotify-player) extension 🙂

Thank you! I was unaware of the property.
I disabled some commands.

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**